src/core/dev/cq_interrupt_distributer.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
+ * Copyright (c) 2001-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef CQ_DISTRIBUTE_H
+#define CQ_DISTRIBUTE_H
+
+#include <sys/mman.h>
+#include <netinet/ip.h>
+#include "doca_error.h"
+#include <doca_buf.h>
+#include <doca_ctx.h>
+#include <doca_log.h>
+#include <doca_dev.h>
+#include <sock/sock-app.h>
+
+// todo - vlogger requires logging from a class - so we define a class
+// when updating to doca logger - get rid of it.
+class cq_interrupt_distributer {
```

Comment:
I agree - but it's not that trivial to put it in a cpp file because of the way we log (macro based-usage).

I've reverted the latest change. I'll open an issue to solve that once we integrate the doca logger api.
(no use to putting an effort in something's that gonna change)

Reply:
See: https://redmine.mellanox.com/issues/4172767